earthaccess/store.py

```diff
@@ -97,8 +96,9 @@ def __init__(self, auth: Any, pre_authorize: bool = False) -> None:
         """
         if auth.authenticated is True:
             self.auth = auth
-            self.s3_fs = None
-            self.initial_ts = datetime.datetime.now()
+            self._s3_credentials: Dict[
+                Tuple, Tuple[datetime.datetime, Dict[str, str]]
+            ] = {}
```

Comment:
I'd like to specify the key `Tuple` a little more strictly if possible. Is it always a 4-tuple? Might be worth having a type alias (`Location = Tuple[Something, SomethingElse, AnotherSomething, SomethingAgain]`). Or maybe better: an annotated namedtuple?

```python
class Location(typing.NamedTuple):
   concept_id: Something
   daac: SomethingAgain
   provider: AnotherSomething
   endpoint: SomethingElse
```

Opinion, not request: I feel giving names to the remaining types may also help with readability, i.e. what can we name `Dict[str, str]` to make it more clear what its purpose is?

Reply:
Given this is (at least currently) only used in one place, and connected to a private attribute, adding additional typing infrastructure like this seems like overkill to me. I'd prefer to wait until this is used in multiple places before adding extra objects that code readers need to know about/keep track of. (Maybe this point is loosely related to the `daac` vs `provider` conversation over in https://github.com/nsidc/earthaccess/issues/349)